src/request-pipeline/index.js

```diff
@@ -168,6 +173,37 @@ function sendResponseHeaders (ctx) {
     ctx.res.addTrailers(ctx.destRes.trailers);
 }
 
+function performWebSocketConnection (ctx) {
```

Comment:
rename to `respondForWebSocket`.

Move `performWebSocketConnection` and `writeWebSocketHead` to the separate file `websocket.js`